include/boost/geometry/algorithms/detail/envelope/range.hpp

```diff
@@ -63,24 +57,29 @@ struct envelope_range
         // initialize MBR
         initialize<Box, 0, dimension<Box>::value>::apply(mbr);
 
-        Iterator it = first;
         if (it != last)
         {
             // initialize box with first element in range
-            dispatch::envelope<value_type>::apply(*it, mbr, strategy);
+            dispatch::envelope
+                <
+                    value_type
+                >::apply(*it, mbr, strategy.get_element_envelope_strategy());
 
             // consider now the remaining elements in the range (if any)
             for (++it; it != last; ++it)
             {
-                dispatch::expand<Box, value_type>::apply(mbr, *it, strategy);
+                dispatch::expand
+                    <
+                        Box, value_type
+                    >::apply(mbr, *it, strategy.get_element_expand_strategy());
```

Comment:
so, `strategy` here is a "envelope_range" strategy that contains an expand strategy for elements. Is this a bit unexpected ? I mean to hide an expand strategy in an envelope strategy.

Reply:
Yes, expand and envelope implementation is tightly bound together. The envelope range has to know how to extract the simple geometries from a range (points in cartesian and segments in spherical/geographic) then how to calculate the bounding boxes from them (e.g. spherical/geographic segment to box) and how to expand by them (e.g. spherical/geographic segment to spherical box and then expansion of spherical box). There are such dependencies in envelope/expand code where one strategy needs lower-level strategies of the other algorithm and this propagates "up". AFAIR I haven't yet refactor the envelope code for Multi-Geometries which adds another layer of dependencies. I left the dependencies as they were before because I wanted to avoid dividing the whole algorithms into strategies one for each kind of Geometry (one for each simple geometry, next for all ranges and next for multi-ranges) but maybe this is how it should be done. But I'd prefer to do it in part 2.